packages/flutter/lib/src/widgets/sliver_explicit_extent_list.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+
+import 'framework.dart';
+import 'scroll_delegate.dart';
+import 'sliver.dart';
+
+/// A sliver that places its box children in a linear array and constrains them
+/// to have the extent returned by [itemExtentBuilder].
+///
+/// _To learn more about slivers, see [CustomScrollView.slivers]._
+///
+/// [SliverExplicitExtentList] arranges its children in a line along
+/// the main axis starting at offset zero and without gaps. Each child is
+/// constrained to the corresponding extent along the main axis
+/// and the [SliverConstraints.crossAxisExtent] along the cross axis.
+///
+/// [SliverExplicitExtentList] is more efficient than [SliverList] because
+/// [SliverExplicitExtentList] does not need to lay out its children to obtain
+/// their extent along the main axis. It's a little more flexible than
+/// [SliverFixedExtentList] because this allow the children to have different extents.
+///
+/// See also:
+///
+///  * [SliverFixedExtentList], whose children are forced to a given pixel
+///    extent.
+///  * [SliverList], which does not require its children to have the same
+///    extent in the main axis.
+///  * [SliverFillViewport], which sizes its children based on the
+///    size of the viewport, regardless of what else is in the scroll view.
+class SliverExplicitExtentList extends SliverMultiBoxAdaptorWidget {
```

Comment:
In general, does this need to be it's own class? Would it be possible to use SliverFixedExtentList for both cases? I am curious for your thinking first, no need to change the code right now. As a developers, I wonder, what is the difference between Fixed and Explicit?

Reply:
I have considered extending SliverFixedExtentList to support variable extent. 
But I'm worried that this will break the developer's understanding of SliverFixedExtentList.
Maybe we need a better name to tell them the difference between the two. 
It's really hard to name it, any suggestions? 😄

Reply:
That's a great name : )